Automattic/wp-calypso
# Framework: Use memoized selectors, and simplify state/posts

This PR shows how we can use ~~[reselect](https://github.com/reactjs/reselect)~~ 'lib/create-selector' (#3545) to create memoized selectors. Why do this? 

* Selectors can compute derived data, allowing Redux to store the minimal possible state.
* Selectors are efficient. A selector is not recomputed unless one of its arguments change.
* Selectors are composable. They can be used as input to other selectors.

Also see: http://redux.js.org/docs/recipes/ComputingDerivedData.html

My primary motivation is to keep our state as minimal as possible, and to move any derived data into memoized selectors. This avoids a lot of complicated use cases in persistence where data may get out of sync with derived data, like a list of posts versus a map of posts indexed by siteId and postId.

This PR removes sitePosts from the redux tree and replaces it with a memoized selector.

## Testing Instructions

The posts tree is currently used in the page editor to back the parent page selector.
* Make sure that your blog has some pages
* Navigate to the editor to add a new page
* Click on Page Options to make sure we can see parent page options
![screen shot 2016-02-22 at 6 01 09 pm](https://cloud.githubusercontent.com/assets/1270189/13239340/4a8ce1e8-d98e-11e5-8a4f-f52a3e9471cc.png)
* Check that the redux tree looks correct
* Editor behaves normally

cc @aduth @blowery @dmsnell @rralian @artpi @retrofox
